src/metabase/pulse/preview.clj

```diff
@@ -0,0 +1,163 @@
+(ns metabase.pulse.preview
+  "Improve the feedback loop for Dashboard Subscription outputs."
+  (:require
+   [clojure.java.io :as io]
+   [clojure.java.shell :as sh]
+   [clojure.string :as str]
+   [hiccup.core :as hiccup]
+   [metabase.email.messages :as messages]
+   [metabase.pulse :as pulse]
+   [metabase.pulse.markdown :as markdown]
+   [metabase.pulse.render :as render]
+   [metabase.pulse.render.image-bundle :as img]
+   [metabase.pulse.render.png :as png]
+   [metabase.pulse.render.style :as style]
+   [toucan2.core :as t2])
+  (:import (java.io File)))
+
+(set! *warn-on-reflection* true)
+
+(defn- os
+  "Returns :win, :mac, :unix, or nil"
+  []
+  (condp
+      #(<= 0 (.indexOf ^String %2 ^String %1))
+      (.toLowerCase (System/getProperty "os.name"))
+      "win" :win
+      "mac" :mac
+      "nix" :unix
+      "nux" :unix
+      nil))
+
+;; taken from https://github.com/aysylu/loom/blob/master/src/loom/io.clj
```

Comment:
This fn and ns also feels very dev-y. Is there a way to move this and the defendpoint into our dev folder and conditionally activate it that way?

Reply:
removed from the preview ns